invoker.js

```diff
@@ -362,13 +362,28 @@ export function apply() {
     const command = invokeEvent.command.toLowerCase();
 
     if (invokee.popover) {
-      const canShow = !invokee.matches(":popover-open");
+      const canShow =
+        !invokee.matches(":popover-open") &&
+        (!state.popoverOpen || command !== "toggle-popover");
       const shouldShow =
         canShow && (command === "toggle-popover" || command === "show-popover");
       const shouldHide = !canShow && command === "hide-popover";
 
       if (shouldShow) {
         invokee.showPopover({ source });
+        // Clicking on the toggle triggers auto-dismiss for the popover, so we need to keep track of the popover state to avoid showing it right back again.
+        state.popoverOpen = true;
+        const unsubscribe = new AbortController();
+        invokee.addEventListener(
+          "toggle",
+          (e) => {
+            if (e.newState === "closed") {
+              state.popoverOpen = false;
+              unsubscribe.abort();
+            }
+          },
+          { signal: unsubscribe.signal },
+        );
```

Comment:
I think `toggle` is probably the wrong thing to use here, because it is queued to a task. You may find that `beforetoggle` is much simpler to integrate with, and may not require the tracking of `state`.

Reply:
I tried using the `beforetoggle` event but couldn't get it working. Is this a blocker for merging this PR?

Reply:
I think it does because of the timing discrepancy.

Reply:
I wonder if a simpler way to tackle this is to check if the command is toggle, and the popover is open? Do we need to track if it is closing?